pkg/controller/common/apm/tracer.go

```diff
@@ -0,0 +1,23 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package apm
+
+import (
+	"github.com/elastic/cloud-on-k8s/pkg/about"
+	"github.com/go-logr/logr"
+	"go.elastic.co/apm"
+)
+
+// NewTracer returns a new APM tracer with the logger in log configured.
+func NewTracer(serviceName string, log logr.Logger) *apm.Tracer {
+	build := about.GetBuildInfo()
+	tracer, err := apm.NewTracer(serviceName, build.Version+"-"+build.Hash)
+	if err != nil {
+		// don't fail the application because tracing fails
+		log.Error(err, "failed to created tracer for "+serviceName)
```

Comment:
What is the way of surfacing any other issues with tracing, ie. bad/missing config? If I never received or stop receiving apm data - how I can detect it and debug it?

Reply:
Assuming we get past this line of code, one line below we are setting a logger into the tracer which reports when data is transmitted to the APM server but will  also report failures: 

```
{"log.level":"info","@timestamp":"2020-01-30T12:46:08.617Z","log.logger":"manager","message":"request failed: request failed with 401 Unauthorized: {\"error\":\"invalid token\"} (next request in ~1s)","service.version":"1.0.0-5791afdd","service.type":"eck","ecs.version":"1.4.0"}                                                                                                                                                   
```

I will change the logger though so that it becomes obvious this error relates to tracing.